core/src/test/java/org/arquillian/smart/testing/scm/git/GitChangeResolverTest.java

```diff
@@ -25,7 +26,12 @@
 public class GitChangeResolverTest {
 
     @Rule
-    public final TemporaryFolder gitFolder = new TemporaryFolder();
+    public final TemporaryFolder gitFolder = new TemporaryFolder(new File("/tmp"));
```

Comment:
`/tmp` is the default location of the temporary directory. Then there are subdirectories created by this rule. There is no need to specify the parent directory. In addition, I believe that the location may change based on the system settings

Reply:
If you run tests using surefire it's setting `java.io.Tmp` to module's target dir due to which all temp folders are creating in `submodule/target`. But JGit's `findGitDir(tempDir)` method is checking for all it's parent for `.git` due to which it got `.git` directory which is of smart-testing git & tests fail.

This is not in IntelliJ as it won't set any system property like surefire

Reply:
Ok, I didn't know that. Thx for the explanation. Bloody surefire...
Well, then we should find another way as in some cases this could fail with wrong permissions. Imagine also the case of using Windows - this path doesn't exist there.

Reply:
The solution is using:
`gitFolder = Files.createTempDirectory("junit-").toFile();` 
 that will be created in `/tmp` by default

Reply:
@MatousJobanek This won't work. Because internally `createTempDirectory` with look for system property `java.io.tmpdir` if parent folder not specified.

In our case we are using jboss-parent as  a parent for `smart-testing/pom.xml` which has `java.io.tmpdir` set to `${project.build.directory}` due to which this api will create tmpdir in current porject's build dir.

Reply:
Is the workaround we discussed offline yesterday working for you?

Reply:
no it's not working. Because We won't get `originial.tmp.dir` as in jboss-parent pom.xml it's defined as `${project.build.directory}`. So it's evaluating ``${project.build.directory}` as original tmp dir

Reply:
I am doing it [this](https://github.com/arquillian/smart-testing/pull/221/files#diff-d0f8b495a0c54a61f1eb2e9f2ed992d4R29) way

Reply:
I've just noticed that when I was trying to fix it locally, I did some additional changes and I was setting it manually to `/tmp` in the code...
But exposing another property via surefire config:
```xml
      <plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <configuration>
          <systemPropertyVariables>
            <original.java.tmp.dir>${java.io.tmpdir}</original.java.tmp.dir>
          </systemPropertyVariables>
        </configuration>
      </plugin>
```
should work. At least, in my case it returns:
```java
"java.io.tmpdir" -> "/home/mjobanek/arquillian/smart-testing/core/target"
"original.java.tmp.dir" -> "/tmp"
```
So technically, it should be also possible to override the settings from the parent and set the same variable to the original one:
```xml
<java.io.tmpdir>${java.io.tmpdir}</java.io.tmpdir>
```